File: doc-testing/src/sdk_strands.ts
import "dotenv/config";
import { init, strandsHooks, shutdown } from "neatlogs";
import { Agent } from "@strands-agents/sdk";
import { OpenAIModel } from "@strands-agents/sdk/models/openai";

async function main() {
  await init({
    apiKey: process.env.NEATLOGS_API_KEY,
    workflowName: "strands-demo",
  });

  const model = new OpenAIModel({ api: "chat", modelId: "gpt-4o" });
  const agent = strandsHooks(new Agent({ model, printer: false }));

  const result = await agent.invoke("In one sentence, what is Strands?");
  const text = result.lastMessage?.content
    ?.map((part) => ("text" in part ? part.text : ""))
    .join("");
  console.log(text);

  await shutdown();
}

main().catch(console.error);

File: agents/src/agents/sdk_strands.py
import os

from dotenv import load_dotenv
import neatlogs
from strands import Agent
from strands.models.openai import OpenAIModel

load_dotenv()

neatlogs.init(api_key=os.environ["NEATLOGS_API_KEY"], workflow_name="strands-demo")

model = OpenAIModel(
    client_args={"api_key": os.environ["OPENAI_API_KEY"]},
    model_id="gpt-4o",
)
agent = Agent(model=model)
neatlogs.strands_hooks(agent)

print(agent("In one sentence, what is Strands?"))

neatlogs.flush()
neatlogs.shutdown()
